src/pke/include/scheme/ckksrns/ckksrns-fhe.h

```diff
@@ -111,6 +111,24 @@ class CKKSBootstrapPrecom {
 
     // coefficients corresponding to conj(U0^T); used in encoding
     std::vector<std::vector<ConstPlaintext>> m_U0hatTPreFFT;
+
+    template <class Archive>
+    void save(Archive& ar) const {
+        ar(cereal::make_nvp("dim1_Enc", m_dim1));
```

Comment:
Why don't we store m_paramsEnc[CKKS_BOOT_PARAMS::GIANT_STEP]) in this case?
From looking at the code, it looks like we use m_dim1 only for the linear case (when level budgets are {1,1}). We use m_paramsEnc for other cases (when level budgets > 1). So I think we should serialize both m_dim1 and m_paramsEnc [CKKS_BOOT_PARAMS::GIANT_STEP].

Reply:
It is redundant to serialize both m_dim1 and m_paramsEnc [CKKS_BOOT_PARAMS::GIANT_STEP], since after deserialization, in EvalBootstrapPrecompute, m_dim1 = dim1[0] is again fed to GetCollapsedFFTParams, which sets m_paramsEnc [CKKS_BOOT_PARAMS::GIANT_STEP] in the same way as it is in EvalBootstrapSetup. The reason why for decoding we serialize m_paramsDec[CKKS_BOOT_PARAMS::GIANT_STEP] is to not create another data member corresponding to dim1[1]. Currently, we serialize the minimum amount of data required for precomputations.

Reply:
I agree that they are redundant but the code for the Linear case currently uses dim1 (some local internal refactoring would be needed). Also, I noticed that the unit tests only for levelBudget = {1,1}. So I am not sure the FFT case is even working now. Worth checking first... Is it even working?

Reply:
If the unit tests already work, then the extra parameter is not needed.

Reply:
i added 4 more unit tests. the old and new ones succeeded with and without serializing/deserializing  m_paramsEnc[CKKS_BOOT_PARAMS::GIANT_STEP]

Reply:
I just looked at the code. It does look like m_dim1 is used there. So no need to serialize the extra parameter.

Reply:
let me commit the new unit tests and run all CIs. i am not adding m_paramsEnc to be serialized for now

Reply:
m_dim1 is serialized from dim1[0] and deserialized, so the correct value is used in the linear case. In the FFT case, m_dim1 = dim1[0] is fed to the same precomputation GetCollapsedFFTParams in both EvalBootstrapSetup and EvalBootstrapPrecompute, so the m_paramsEnc are set correctly.

Alternatively, a redundant but visually simpler way is to only compute m_paramsEnc and m_paramsDec once in EvalBootstrapSetup and serialize them, along with m_dim1, and not recompute in EvalBootstrapPrecompute.

Reply:
i have the new unit tests committed and started CI jobs